client/authority-discovery/src/error.rs

```diff
@@ -28,6 +28,8 @@ pub enum Error {
 	ReceivingDhtValueFoundEventWithDifferentKeys,
 	/// Received dht value found event with no records.
 	ReceivingDhtValueFoundEventWithNoRecords,
+	/// Received dht value found event even though querying for authorities is disabled.
+	ReceivingDhtValueFoundEventWithQueryingDisabled,
```

Comment:
This assumes that the authority discovery is the only code that will ever make DHT queries.
This cannot happen at the moment, but if another module calls `get_value` on the network service to make its own queries (unrelated to the authority discovery), these errors would be triggered.